### DISCUSSION TITLE
ellc error when running inverse paper examples

### DISCUSSION THREAD

**Discussion by astro-sharp** [phoebe-project/phoebe2#817] [Questions]
Hi,

I'm very new to the amazing Phoebe 2.4 and I'm finding learning curves everywhere I look! I'd appreciate help with this issue I cannot seem to solve...

When running the inverse paper examples (http://phoebe-project.org/docs/2.4/examples/inverse_paper_examples) I get an error involving the ellc module. I have managed to 'pip install ellc' ok and I can 'import ellc' with no errors.

![image](https://github.com/phoebe-project/phoebe2/assets/55244395/3286582a-eeee-4ea4-851b-ae9b03c1366f)

The error I get is :
"ImportError: ellc failed to load required lc and/or rv functions with error: DLL load failed while importing ellc_f: The specified module could not be found."

This occurs when I call:

b.run_compute(compute='fastcompute')

Many thanks
Ian

**Comment by kecnry** [comment]
Welcome! There are definitely plenty of learning curves, but hopefully once you get comfortable with the basic concepts, things will start feeling more intuitive 🤞   

Can you do `from ellc import lc` outside of phoebe successfully?  I'm guessing `import ellc` alone does not pull in the `lc` submodule so might work, but then later fail when trying to access `lc` (which is needed by phoebe). 

Assuming that fails, then the problem is with the installation of ellc on your system (from the error message, it looks like the compiled fortran files are the issue).  It looks like there are a few similar issues posted in the ellc repository that you might want to check out and see if any of the advice there helps in your case, or you can try opening a new issue there.

**Reply by astro-sharp** [reply]
Many thanks for your prompt reply, especially in the holiday season!

Yes, I should have mentioned I get the same error if I try `from ellc import lc` (or rv).

I had to install a FORTRAN compiler to install ellc as I didn't have one, so I installed MinGW and added the FORTRAN compiler in the installation manager. Once added to the PATH, it enabled me to `pip install ellc` without any errors.

I will check out the ellc repository as you suggest...

Ian.

**Reply by kecnry** [reply]
I'll also just add that `ellc` is a separate code that phoebe interfaces with as an "alternate backend", but is not required.  It is very handy in scenarios where its assumptions are "safe" for a particular system as it is an efficient way to use optimizers, etc, by temporarily bypassing the expensive aspects of the native phoebe backend.  But if you don't think you'll need to use it, feel free to just swap out with another backend and the general principles in that tutorial will still apply.  See this [table comparing the different supported forward-model backends](http://phoebe-project.org/docs/latest/examples/compute_comparison_table) for more details.

**Reply by astro-sharp** [reply]
Thanks again Kyle,

I'll let you know how I get on...

Cheers
Ian.
